File: KuantumKaosYonetimi/CSharp/KuantumKaosYonetimi/KuantumNesnesi.cs
namespace KuantumKaosYonetimi
{
    // A. Temel Yapı: Soyut Sınıf (Abstract Class)
    public abstract class KuantumNesnesi
    {
        // Özellikler (Properties)
        public string ID { get; }
        public int TehlikeSeviyesi { get; }

        // Stabilite (Kapsülleme)
        private double _stabilite;
        public double Stabilite
        {
            get => _stabilite;
            set
            {
                // Kapsülleme: 0-100 aralığı dışındaki değerleri engelle
                if (value > 100)
                {
                    _stabilite = 100;
                }
                else if (value < 0)
                {
                    // Eğer stabilitenin düşmesi 0'ın altına inerse, hatayı fırlatmak için
                    // sadece değeri 0 olarak ayarlarız. Hatanın fırlatılması main döngüde kontrol edilir.
                    _stabilite = 0;
                }
                else
                {
                    _stabilite = value;
                }
            }
        }

        // Yapıcı Metot (Constructor)
        public KuantumNesnesi(string id, int tehlikeSeviyesi, double stabilite)
        {
            ID = id;
            TehlikeSeviyesi = tehlikeSeviyesi;
            Stabilite = stabilite; // Kapsüllenmiş setter'ı kullanır
        }

        // Soyut Metot (Abstract Method)
        public abstract void AnalizEt();

        // Somut Metot
        public string DurumBilgisi()
        {
            return $"[ID: {ID}] - Durum: Stabilite = {Stabilite:F2}% (Tehlike: {TehlikeSeviyesi}/10)";
        }
    }
}

File: KuantumKaosYonetimi/Java/KuantumKaosYonetimi/KuantumNesnesi.java
// KuantumNesnesi.java
package KuantumKaosYonetimi;

// Abstract sınıf bildirimi
public abstract class KuantumNesnesi {
    
    // Özellikler (Fields)
    private final String ID;
    private final int TehlikeSeviyesi;
    private double Stabilite; // Kapsülleme ile yönetilecek
    
    // Yapıcı Metot (Constructor)
    public KuantumNesnesi(String id, int tehlikeSeviyesi, double stabilite) {
        this.ID = id;
        this.TehlikeSeviyesi = tehlikeSeviyesi;
        // Setter metodunu kullanarak kapsüllemeyi çalıştırır
        setStabilite(stabilite); 
    }

    // Stabilite Setter (Kapsülleme)
    public void setStabilite(double stabilite) {
        if (stabilite > 100) {
            this.Stabilite = 100;
        } 
        // 0'dan küçük değerler için fırlatma kontrolü Main döngüde yapılacak
        else if (stabilite < 0) {
            this.Stabilite = 0;
        } 
        else {
            this.Stabilite = stabilite;
        }
    }

    // Stabilite Getter
    public double getStabilite() {
        return Stabilite;
    }

    // ID Getter (ID final olduğu için setter'a gerek yok)
    public String getID() {
        return ID;
    }
    
    // TehlikeSeviyesi Getter
    public int getTehlikeSeviyesi() {
        return TehlikeSeviyesi;
    }

    // AnalizEt(): Soyut Metot
    public abstract void AnalizEt();

    // DurumBilgisi(): Nesnenin ID'sini ve o anki stabilitesini string döndürür
    public String DurumBilgisi() {
        // String.format kullanarak C# stili formatlama yapılır
        return String.format("[ID: %s] - Durum: Stabilite = %.2f%% (Tehlike: %d/10)", 
                              this.ID, this.Stabilite, this.TehlikeSeviyesi);
    }
}